mlx/backend/cuda/compiled.cpp

```diff
@@ -227,6 +291,7 @@ void Compiled::eval_gpu(
 
   auto kernel = mod.get_kernel(kernel_name);
   auto [num_blocks, block_dims] = get_launch_args(kernel, outputs[0], large);
```

Comment:
`get_launch_args` takes `work_per_thread` as the last variable. Maybe be better to use than dividing `num_blocks.x` for the large kernels.